elasticip_exoscale.go

```diff
@@ -162,115 +189,44 @@ func (r *exoscaleElasticIPRefresher) Logger() *logrus.Entry {
 }
 
 func (r *exoscaleElasticIPRefresher) Refresh(ctx context.Context) error {
-	// Find virtual machines with IP address
-	vms, err := r.client.ListWithContext(ctx, &egoscale.VirtualMachine{
-		Nic: []egoscale.Nic{
-			egoscale.Nic{
-				IsDefault: true,
-				IPAddress: r.network.IP,
-			},
-		},
-	})
+	err := r.client.AttachInstanceToElasticIP(ctx, r.zone, r.instance, r.eip)
 	if err != nil {
-		return err
+		return fmt.Errorf("while attaching the IP to this instance: %s", err)
 	}
+	logrus.Infof("Ensured that %s is attached to instance %s", r.eip.IPAddress, *r.instance.ID)
 
-	commands := []egoscale.Command{
-		// Always force association to target machine
-		&egoscale.AddIPToNic{
-			NicID:     r.nicID,
-			IPAddress: r.network.IP,
-		},
+	vms, err := r.client.ListInstances(ctx, r.zone)
+	if err != nil {
+		return fmt.Errorf("Unable to list instances: %s", err)
 	}
 
-	// Determine which other virtual machines have the IP address associated
-	// and prepare commands to disassociate
-	for _, i := range vms {
-		vm := i.(*egoscale.VirtualMachine)
-		nic := vm.DefaultNic()
-
-		r.logger.WithFields(logrus.Fields{
-			"vm-id":   vm.ID,
-			"vm-name": vm.Name,
-			"vm-nic":  nic,
-		}).Debugf("Virtual machine %q", vm.Name)
-
-		if nic.ID != nil && r.nicID.Equal(*nic.ID) {
-			// Desired target
+	// Detach from other instances
+	var detacherrs error
+	for _, vm := range vms {
+		if *vm.ID == *r.instance.ID {
```

Comment:
Please recheck, if it's not a comparison of pointers https://github.com/vshn/floaty/commit/5a5d16edf74f6858774b8eaa16c0f6737d9de33b

Reply:
IDs are `*string` in the v2 API, so we're comparing strings here.

Reply:
Perfect.